client/profile-wizard/steps/product-types/index.js

```diff
@@ -57,17 +58,41 @@ export class ProductTypes extends Component {
 	}
 
 	onContinue() {
+		const { selected } = this.state;
+		const { installedPlugins = [] } = this.props;
+
 		if ( ! this.validateField() ) {
 			return;
 		}
 
-		const { createNotice, goToNextStep, updateProfileItems } = this.props;
+		const {
+			createNotice,
+			goToNextStep,
+			installAndActivatePlugins,
+			updateProfileItems,
+		} = this.props;
 
 		recordEvent( 'storeprofiler_store_product_type_continue', {
-			product_type: this.state.selected,
+			product_type: selected,
 		} );
 
-		updateProfileItems( { product_types: this.state.selected } )
+		if (
+			window.wcAdminFeatures &&
+			window.wcAdminFeatures.subscriptions &&
+			! installedPlugins.includes( 'woocommerce-payments' ) &&
+			selected.includes( 'subscriptions' )
+		) {
+			installAndActivatePlugins( [ 'woocommerce-payments' ] )
```

Comment:
I think we should either make use of `promise.all` with the `updateProfileItems` call on line 95, or nest them. We do something similar in the `business-details` step, [see here](https://github.com/woocommerce/woocommerce-admin/blob/main/client/profile-wizard/steps/business-details/flows/selective-bundle/index.js#L100-L131).

Suggesting this, as for me it already continued to the next step and the `continue` button still showed as busy as wc pay hadn't finished installing yet, this felled a bit weird.

Reply:
I'm not sure about the `promise.all` approach since the user will have to wait until the WC-Pay plugin be installed and activated before continuing. I wanted to have that process underneath while the user continues filling the OBW.

> as for me it already continued to the next step and the continue button still showed as busy as wc pay hadn't finished installing yet, this felled a bit weird.

Yes, the `Continue` button is shown as `busy` because the plugin is being installed, I didn't feel it as a problem, but what you said makes sense.
I'm not sure which is the best approach here but if you're confident with the `promise.all` I can add it.

Reply:
I would still recommend the `promise.all` mostly because we already doing it in the free features step.

If you do want to keep it in the background, I think the `Continue` button shouldn't reflect the installation.